lib/config.ex

```diff
@@ -1,28 +1,144 @@
 defmodule Xcribe.Config do
+  @moduledoc """
+  Handle Xcribe configurations.
+
+  You must configure Xcribe in your test config file `config/test.exs` as:
+      config: xcribe, :configuration, [
+        information_source: YourApp.YouModuleInformation,
+        format: :swagger,
+        output: "app_doc.json",
+        env_var: "CI_ENV_FOR_DOC",
+        json_library: Jason
+      ]
+
+      :api_blueprint -> "api_doc.apib"
+      :swagger -> "openapi.json"
+
+  ### Available configurations:
+    * `:information_source` - Module that implements `Xcribe.Information` with
+    API information. It's required.
+    * `:output` - The name of file output with generated configuration. Default
+    value changes by the format, 'api_blueprint.apib' for Blueprint and
+    'app_doc.json' for swagger.
+    * `:format` - Format to generate documentation, allowed `:api_blueprint` and
+    `:swagger`. Default `:api_blueprint`.
+    * `:env_var` - Environment variable name for active Xcribe documentation
+    generator. Default is `XCRIBE_ENV`.
+    * `:json_library` - The library to be used for json decode/encode (Jason
+    and Poison are supported). The default is the same as `Phoenix` configuration.
+  """
+
+  alias Xcribe.{MissingInformationSource, UnknownFormat}
+
   @valid_formats [:api_blueprint, :swagger]
-  def output_file, do: Application.get_env(:xcribe, :output_file, default_output_file())
 
+  @doc """
+  Return the file name to output generated documentation.
+
+  If no config was given the default names are `api_doc.apib` for Blueprint
+  format and `openapi.json` for Swagger format.
+
+  To configure output name:
+
+      config :xcribe, :configuration, [
+        output: "custom_name.json"
+      ]
+  """
+  def output_file, do: get_xcribe_config(:output, default_output_file())
+
+  @doc """
+  Return the format for documentation.
+
+  Default is `:api_blueprint`. If an invalid format is given an `Xcribe.UnknownFormat`
+  exception will raise.
+
+  To configure the documentation format:
+
+      config :xcribe, :configuration, [
+        format: :swagger
+      ]
+  """
   def doc_format do
-    :xcribe
-    |> Application.get_env(:doc_format, :api_blueprint)
+    :format
```

Comment:
this changing breaks compatibility, maybe we could maintain the old name with a deprecation warning or bump a major version to avoid automatic breaking updates